CONTRIBUTING.md

```diff
@@ -188,7 +188,70 @@ with one another are rolled up.
 Speaking of tests, Rust has a comprehensive test suite. More information about
 it can be found [here][rctd].
 
-### External Dependencies
+### External Dependencies (subtree)
+
+As a developer to this repository, you don't have to treat the following external projects
+differently from other crates that are directly in this repo:
+
+* none so far, see https://github.com/rust-lang/rust/issues/70651 for more info
+
+They are just regular files and directories. This is in contrast to `submodule` dependencies
+(see below for those). Only tool authors will actually use any operations here.
+
+#### Synchronizing a subtree
+
+There are two synchronization directions: `subtree push` and `subtree pull`.
+
+```
+git subtree push -P src/tools/clippy git@github.com:your-github-name/rust-clippy sync-from-rust
+```
+
+takes all the changes that
+happened to the copy in this repo and creates commits on the remote repo that match the local
+changes. Every local commit that touched the subtree causes a commit on the remote repo, but is
+modified to move the files from the specified directory to the tool repo root.
+
+Make sure to not pick the `master` branch on the tool repo, so you can open a normal PR to the tool
+to merge that subrepo push.
```

Comment:
It seems to me that the important thing is to not have `git@github.com:rust-lang/rust-clippy.git` as the repository URL, i.e., you should never be pushing into branches there. But which branch you choose in your fork is largely irrelevant (you can push to master if you like).

Reply:
I don't have my own fork of clippy and miri, should I have?

Reply:
I think it is better to keep personal feature branches out of the main repo. Not a strong opinion though.

Reply:
@oli-obk Not to mention that you can only do this because of your powers on those repos, most contributors *have to* use a fork.

Reply:
IMO for repos with many contributors the amount of feature branches in the main repo should be kept as low as possible. Clippy has even an issue, where a contributor complained about the amount of feature branches: rust-lang/rust-clippy#4745

Reply:
For the record, the 
`git subtree push -P src/tools/clippy git@github.com:your-github-name/rust-clippy sync-from-rust`  command is just crashing on my machine
```
git subtree push -P src/tools/clippy git@github.com:matthiaskrgr/rust-clippy2  sync-from-rust

git push using:  git@github.com:matthiaskrgr/rust-clippy2 sync-from-rust
/usr/lib/git-core/git-subtree: line 757: 171266 Done                    eval "$grl"
     171267 Segmentation fault      (core dumped) | while read rev parents; do
    process_split_commit "$rev" "$parents" 0;
done
```